File: strings/string_fundamental/python/palindrome.py
"""
 A palindrome is a word, phrase, or sequence that reads the same backward as forward, e.g., madam or nurses run.
"""

# Function to check if string is a palindrome
def check_palindrome(str):
    str = str.lower()
    rev = str[::-1]
    if (str == rev):
        return True
    else:
        return False

    

if __name__ == '__main__':
    string_one = input("Enter string to check if it's a palindrome:")
    is_p = check_palindrome(string_one)
    if is_p:
        print("The string is a palindrome")
    else:
        print("The string is not an palindrome.")

File: strings/string_fundamental/go/palindrome.go
//Translation of Python implementation. See README for details of what this is supposed to do.
package main

import (
	"bufio"
	"fmt"
	"os"
	"strings"
)

//Returns True if input string is a palindrome
func check_palindrome(str string) bool {
	//First, remove uppercase characters and whitespace
	str = strings.Join(strings.Fields(strings.ToLower(str)), "")

	//Create a second string that is the mirror image of the first
	rev := reverse(str)

	return str == rev
}

//Flip the order of the charaters in a string
func reverse(s string) string {
	characters := []rune(s)
	start, end := 0, len(s)-1
	for ; start < end; start, end = start+1, end-1 {
		characters[start], characters[end] = characters[end], characters[start]
	}
	return string(characters)
}

func main() {
	getline := bufio.NewReader(os.Stdin)
	fmt.Print("Enter text: ")
	input, _ := getline.ReadString('\n')
	is_p := check_palindrome(input)
	if is_p {
		fmt.Println("The string is a palindrome")
	} else {
		fmt.Println("The string is not an palindrome.")
	}
}
